src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/memberdays/MemberDaysReport.java

```diff
@@ -47,41 +41,28 @@ public MemberDaysReportData runReport(ListingDetails listings, List<SubmitEvent>
     }
 
     private MemberDaysReportData initiateReport(ListingDetails listingDetails) {
-        String caseTypeId = listingDetails.getCaseTypeId();
-        var office = UtilHelper.getListingCaseTypeId(caseTypeId);
+        var office = UtilHelper.getListingCaseTypeId(listingDetails.getCaseTypeId());
         var reportData = new MemberDaysReportData();
+        var caseData = listingDetails.getCaseData();
         reportData.setOffice(office);
-        reportData.setDurationDescription(getDurationText(listingDetails.getCaseData()));
-        reportData.setHearingDateType(listingDetails.getCaseData().getHearingDateType());
+        reportData.setHearingDateType(caseData.getHearingDateType());
         reportData.setReportType(MEMBER_DAYS_REPORT);
         reportData.setDocumentName(MEMBER_DAYS_REPORT);
+        reportData.setListingDate(caseData.getListingDate());
+        reportData.setListingDateFrom(caseData.getListingDateFrom());
+        reportData.setListingDateTo(caseData.getListingDateTo());
         return reportData;
     }
 
-    private String getDurationText(ListingData currentCaseData) {
-        var description = "";
-
-        if (currentCaseData.getHearingDateType().equals(SINGLE_DATE_HEARING_REPORT)) {
-            description = "On " + currentCaseData.getListingDate();
-        } else if (currentCaseData.getHearingDateType().equals(DATE_RANGE_HEARING_REPORT)) {
-            description = "Between " + currentCaseData.getListingDateFrom()
-                    + " and " + currentCaseData.getListingDateTo();
-        }
-
-        return description;
-    }
-
     private void addReportDetails(MemberDaysReportData reportData, List<SubmitEvent> submitEvents,
                                   ListingData listingData) {
         List<MemberDaysReportDetail> interimReportDetails = new ArrayList<>();
+
         for (var submitEvent : submitEvents) {
-            if (!isValidCase(submitEvent)) {
+            if (submitEvent.getCaseData().getHearingCollection().isEmpty()) {
```

Comment:
What happens if hearingCollection is null? I think you will get Exception here. 

May be use CollectionUtils.IsEmpty(submitEvent.getCaseData().getHearingCollection()) which will check for null as well?

Reply:
Yes you are right. Fixed.